app/js/lib/foxbox/recipes.js

```diff
@@ -24,232 +49,222 @@ export default class Recipe {
+
+          selectors.enabledSelectors.push({ id: service.getEnabled });
+          selectors.sourceSelectors.push({ id: service.getSource });
+
+          return selectors;
+        },
+        { enabledSelectors: [], sourceSelectors: [] }
+      );
+
+      const getterURL = `${this[p.net].origin}/api/` +
+        `v${this[p.settings].apiVersion}/channels/get`;
+
+      return Promise.all([
+        this[p.net].fetchJSON(getterURL, 'PUT', enabledSelectors),
+        this[p.net].fetchJSON(getterURL, 'PUT', sourceSelectors)
+      ])
+      .then(([statuses, sources]) => {
+        return services.map(service => {
+          const statusResponse = statuses[service.getEnabled];
+          const sourceResponse = sources[service.getSource];
+
+          if (statusResponse && statusResponse.OnOff) {
+            service.status = statusResponse.OnOff === 'On';
+          } else {
+            console.error(
+              'Error occurred while retrieving recipe (%s) status: ',
+              service.id,
+              statusResponse && statusResponse.Error
+            );
+            service.status = false;
           }
 
-          return service;
-        }
-      ))
-      // User friendly getters.
-      .then(services => services.map(service => {
-          switch (service.adapter) {
-            case 'clock@link.mozilla.org':
-              service.getters = [
-                {
-                  label: 'in the morning',
-                  value: 'service:clock@link.mozilla.org,CurrentTimeOfDay,eq,' +
-                  '08:00am'
-                },
-                {
-                  label: 'in the afternoon',
-                  value: 'service:clock@link.mozilla.org,CurrentTimeOfDay,eq,' +
-                  '02:00pm'
-                },
-                {
-                  label: 'in the evening',
-                  value: 'service:clock@link.mozilla.org,CurrentTimeOfDay,eq,' +
-                  '06:00pm'
-                }
-              ];
-              break;
-
-            default:
-              //service.getters = service.getters;
-              break;
+          if (sourceResponse && sourceResponse.String) {
+            service.source = JSON.parse(sourceResponse.String);
+          } else {
+            console.error(
+              'Error occurred while retrieving recipe (%s) source: ',
+              service.id,
+              sourceResponse && sourceResponse.Error
+            );
+            service.source = null;
           }
 
-          return service;
-        }
-      ))
-      // Map all the recipes to a more user-friendly format.
-      .then(services => services.map(service => ({
-        id: service.id,
-        label: service.name,
-        get: service.getters
-      })));
+          return new Recipe(service);
+        });
+      });
+    });
   }
 
-  getServicesWithSetters() {
+  getGetters() {
+    // Currently we support only Clock as a trigger.
+    const supportedKinds = ['CurrentTimeOfDay'];
```

Comment:
If you don't mind, for the demo I'd rather indicate services that we support, instead of filtering out unsupported ones, I've just found it a bit easier to add new supported services. I guess the next one will be motion sensor